speclang_modern.tex

```diff
@@ -2543,7 +2546,8 @@ \subsubsection{Allocation clauses for function
 \\
 For any pointer \lstinline|p| and label \lstinline|L|
 \begin{listing-nonumber}
-\freeable{L}(p) <==> (\allocation{L}(p)==\dynamic && p==\base_addr{L}(p)).
+\freeable{L}(p) <==> (\allocation{L}(p)==\dynamic
+                        && (void*)p==\base_addr{L}(p))
```

Comment:
I don't have a strong position on this, but couldn't `\base_addr` return a `char *` (so that it would be possible to say e.g. `\valid(\base_addr(p) + (0 .. \block_length(p)))` without cast). Looking at C standard, `unsigned char*` might make even more sense: the _object representation_ of any value is an array of `unsigned char`, as per 6.2.6.1§4

Reply:
Accepted since I have no strong opinion on the thing and both are correct.